networkit/cpp/flow/EdmondsKarp.cpp

```diff
@@ -55,22 +56,125 @@ edgeweight EdmondsKarp::BFS(std::vector<edgeweight> &residFlow, std::vector<node
     return 0.0;
 }
 
+edgeweight EdmondsKarp::directedBFS(const std::vector<count> &reverseEdges,
+                                    std::vector<node> &pred) const {
+    std::fill(pred.begin(), pred.end(), none);
+    pred.resize(graph->upperNodeIdBound(), none);
+    std::vector<edgeweight> gain(graph->upperNodeIdBound(), std::numeric_limits<edgeweight>::max());
+
+    std::queue<node> Q;
+    Q.push(source);
+    pred[source] = source;
+    gain[source] = std::numeric_limits<edgeweight>::max();
+    ;
+
+    while (!Q.empty()) {
```

Comment:
Nit: I suggest to change this with a `do { ... } while (!Q.empty());`. `Q` is never empty in the first iteration.